src/main/java/core/mate/academy/service/MachineServiceImpl.java

```diff
@@ -1,7 +1,54 @@
 package core.mate.academy.service;
 
-/**
- * Your implementation of MachineService.
- */
-public class MachineServiceImpl {
+import core.mate.academy.model.Bulldozer;
+import core.mate.academy.model.BulldozerProducer;
+import core.mate.academy.model.Excavator;
+import core.mate.academy.model.ExcavatorProducer;
+import core.mate.academy.model.Machine;
+import core.mate.academy.model.Truck;
+import core.mate.academy.model.TruckProducer;
+import java.util.ArrayList;
+import java.util.List;
+
+public class MachineServiceImpl implements MachineService<Machine> {
+
+    @Override
+    public List<Machine> getAll(Class<? extends Machine> classObject) {
+        List<? extends Machine> machines;
+        if (classObject.equals(Bulldozer.class)) {
+            machines = new BulldozerProducer().get();
```

Comment:
```suggestion
            machines = bulldozerProducer.get();
```
lets create class-level variables for each producer to avoid creating a new instance each time this method is invoked